import-export-cli/impl/changeAPIStatus.go

```diff
@@ -0,0 +1,112 @@
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package impl
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/go-resty/resty"
+	"github.com/wso2/product-apim-tooling/import-export-cli/credentials"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+)
+
+// ChangeAPIStatusInEnv function is used with change-status api command
+func ChangeAPIStatusInEnv(credential credentials.Credential, accessToken, environment, stateChangeAction, name, version, provider string) (*resty.Response, error) {
+	changeAPIStatusEndpoint := utils.GetApiListEndpointOfEnv(environment, utils.MainConfigFilePath)
+	return changeAPIStatus(changeAPIStatusEndpoint, stateChangeAction, name, version, provider, environment, accessToken, credential)
+}
+
+// changeAPIStatus
+// @param changeAPIStatusEndpoint : API Manager Publisher REST API Endpoint for the environment
+// @param stateChangeAction : Action to be performed to change the state of the API
+// @param name : Name of the API
+// @param version : Version of the API
+// @param provider : Provider of the API
+// @param environment : Environment where the API resides
+// @param accessToken : Access Token for the resource
+// @param credential : Credentials of the logged-in user
+// @return response Response in the form of *resty.Response
+func changeAPIStatus(changeAPIStatusEndpoint, stateChangeAction, name, version, provider, environment, accessToken string, credential credentials.Credential) (*resty.Response, error) {
+	changeAPIStatusEndpoint = utils.AppendSlashToString(changeAPIStatusEndpoint)
+	apiId, err := getAPIIdForStateChange(accessToken, name, version, provider, environment, credential)
+	if err != nil {
+		utils.HandleErrorAndExit("Error while getting API Id for state change ", err)
+	}
+	url := changeAPIStatusEndpoint + "change-lifecycle?action=" + stateChangeAction + "&apiId=" + apiId
+	utils.Logln(utils.LogPrefixInfo+"APIStateChange: URL:", url)
+	headers := make(map[string]string)
+	headers[utils.HeaderContentType] = utils.HeaderValueApplicationJSON
+	headers[utils.HeaderAuthorization] = utils.HeaderValueAuthBearerPrefix + " " + accessToken
+
+	resp, err := utils.InvokePOSTRequest(url, headers, "")
+
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// Get the ID of an API if available
+// @param accessToken : Token to call the Publisher Rest API
+// @param name : Name of the API
+// @param version : Version of the API
+// @param provider : Provider of the API
+// @param environment : Environment where the API resides
+// @return apiId, error
+func getAPIIdForStateChange(accessToken, name, version, provider, environment string, credential credentials.Credential) (string, error) {
+	// Unified Search endpoint from the config file to search APIs
+	unifiedSearchEndpoint := utils.GetUnifiedSearchEndpointOfEnv(environment, utils.MainConfigFilePath)
```

Comment:
Do we need a special method for this? I remember there were several number of methods to get API. Its better to move those similar methods to a single package and make it reusable. I guess this method does nothing more than getting the ID of the API by doing a search from name and version?

Reply:
There is a method called getAPIID in impl.importAPI.go.. Can we make it public and reuse it instead?

Also, its better to keep single responsibility across the files then we can easily find methods we need and can make it easily reusable. We don't usually expect a public method GetAPIID inside changeAPIStatus.go or importAPI.go. A better place would be having it in a file similar to apis.go or api.go.

Reply:
There is another getAPIId method inside deleteAPI.go

Reply:
Fixed via https://github.com/wso2/product-apim-tooling/pull/518/commits/d268de6195b81448eb0d7a780e0610316ab2b37b, https://github.com/wso2/product-apim-tooling/pull/518/commits/944b4e515b1372bf691e8574e4c58484b55aeece and https://github.com/wso2/product-apim-tooling/pull/518/commits/f5cb477f64ff9f04f524065a00cd3a68be80d97a

Reply:
Great!